src/librustdoc/html/format.rs

```diff
@@ -718,30 +735,44 @@ impl fmt::Display for clean::Type {
 }
 
 fn fmt_impl(i: &clean::Impl, f: &mut fmt::Formatter, link_trait: bool) -> fmt::Result {
+    let mut plain = String::new();
+
     if f.alternate() {
         write!(f, "impl{:#} ", i.generics)?;
     } else {
         write!(f, "impl{} ", i.generics)?;
     }
+    plain.push_str(&format!("impl{:#} ", i.generics));
```

Comment:
Could this just be:

```rust
let mut plain = format!("impl{:#} ", i.generics);
```

...and then getting rid of the `String::new()` above?